engine/src/main/java/org/terasology/engine/core/bootstrap/EntitySystemSetupUtil.java

```diff
@@ -165,7 +166,7 @@ private static EventSystem createEventSystem(NetworkSystem networkSystem, PojoEn
 
     private static void registerComponents(ComponentLibrary library, ModuleEnvironment environment) {
         for (Class<? extends Component> componentType : environment.getSubtypesOf(Component.class)) {
-            if (componentType.getAnnotation(DoNotAutoRegister.class) == null) {
+            if (componentType.getAnnotation(DoNotAutoRegister.class) == null && !componentType.isInterface() && !Modifier.isAbstract(componentType.getModifiers())) {
```

Comment:
why is this change necessary?

Reply:
Because:
1. `@DoNotAutoRegister` is part of TS, not gestalt.
2.  It fails with `EmptyComponent`. really abstract classes is not functional for ECS.

Reply:
Long line warning, add line break.